### PR TITLE
Use GitHub actions

### PR DESCRIPTION
**Pull request by carltongibson**
* Adds a tox config and GitHub Actions workflow. 
* Uses flake8, fixes errors.
* Removed travis config. 

Possibly: could drop Python 3.5 support, as it is EOL. (But no necessity for that.)

### CODE REVIEW

**Comment by carltongibson** [comment]
Checks are running on my fork until merged. 

https://github.com/carltongibson/asgiref/actions/runs/348143120

**Review by andrewgodwin — APPROVED** [review]

**Comment by andrewgodwin** [comment]
Looks good - I'll merge it and if it all runs smoothly for a few commits, we can yank out Travis.

**Pull request merged**
